### DISCUSSION TITLE
GRPC reflection

### DISCUSSION THREAD

**Discussion by megasuperlexa** [AnWeber/vscode-httpyac#253] [Q&A]
Is there a way to enable it globally (like in Rider) and not on a per-endpoint basis?

**Comment by AnWeber** [comment]
How did rider solve it? Unfortunately I can't find any documentation.

I have currently implemented it in such a way that you have to explicitly specify that the server supports GRPC Refleciton. However, this can also be done in a global Http region.

```
# @grpc-reflection grpc.postman-echo.com

###
GRPC grpc.postman-echo.com/HelloService/sayHello
{
  "greeting": "world"
}

###
GRPC grpc.postman-echo.com/HelloService/sayHello
{
  "greeting": "world"
}

```

it would also be possible to outsource it to a separate plugin, but I would find the same solution as Rider/ Intellij more charming.

**Comment by megasuperlexa** [comment]
> How did rider solve it? Unfortunately I can't find any documentation.

It just works, I did not turn any knobs. I guess it just probes for reflection in case no proto is available (httpYac simply complains "no proto" at this point)

**Comment by AnWeber** [comment]
with next Release GRPC Reflection is used in every call
